sql/tests: TestSchemaChangesInParallel failed

sql/tests.TestSchemaChangesInParallel [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9110839?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9110839?buildTab=artifacts#/) on master @ [0a4b38348c6d4874a619126e5273829d7a5bc894](https://github.com/cockroachdb/cockroach/commits/0a4b38348c6d4874a619126e5273829d7a5bc894):


```
I230317 11:36:49.006882 98463 sql/gcjob/gc_job_utils.go:296  [T1,n1,job=SCHEMA CHANGE GC id=848634212708319233] 2948  updated running status: waiting for GC TTL
I230317 11:36:49.015073 98416 5@util/log/event_log.go:32  [T1,n1,job=NEW SCHEMA CHANGE id=848634212651204609] 2949 ={"Timestamp":1679053008933446073,"EventType":"finish_schema_change","InstanceID":1,"DescriptorID":109}
I230317 11:36:49.015610 98618 sql/gcjob/gc_job_utils.go:289  [T1,n1,job=SCHEMA CHANGE GC id=848634212798824449] 2950  updated progress payload: tables:<id:114 > tables:<id:118 > ranges_unsplit_done:true
I230317 11:36:49.020716 98463 sql/gcjob/gc_job_utils.go:289  [T1,n1,job=SCHEMA CHANGE GC id=848634212708319233] 2951  updated progress payload: tables:<id:113 > tables:<id:116 > ranges_unsplit_done:true
I230317 11:36:49.044647 98463 sql/gcjob/gc_job_utils.go:296  [T1,n1,job=SCHEMA CHANGE GC id=848634212708319233] 2955  updated running status: deleting data
I230317 11:36:49.046147 98618 sql/gcjob/gc_job_utils.go:296  [T1,n1,job=SCHEMA CHANGE GC id=848634212798824449] 2956  updated running status: waiting for GC TTL
I230317 11:36:49.024450 98771 jobs/adopt.go:247  [T1,n1] 2952  job 848634213013553153: resuming execution
I230317 11:36:49.056135 98463 sql/gcjob/table_garbage_collection.go:208  [T1,n1,job=SCHEMA CHANGE GC id=848634212708319233] 2957  deleting data for table 113
I230317 11:36:49.024527 98689 jobs/adopt.go:247  [T1,n1] 2953  job 102: resuming execution
E230317 11:36:49.059817 98689 jobs/adopt.go:211  [T1,n1] 2958  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230317 11:36:49.024547 98770 jobs/adopt.go:247  [T1,n1] 2954  job 848634212156407809: resuming execution
I230317 11:36:49.062146 98463 sql/gcjob/table_garbage_collection.go:208  [T1,n1,job=SCHEMA CHANGE GC id=848634212708319233] 2959  deleting data for table 116
I230317 11:36:49.065296 98618 sql/gcjob/gc_job_utils.go:289  [T1,n1,job=SCHEMA CHANGE GC id=848634212798824449] 2960  updated progress payload: tables:<id:114 > tables:<id:118 > ranges_unsplit_done:true
I230317 11:36:49.070293 98618 sql/gcjob/gc_job_utils.go:296  [T1,n1,job=SCHEMA CHANGE GC id=848634212798824449] 2961  updated running status: deleting data
I230317 11:36:49.070420 98463 sql/gcjob/gc_job_utils.go:289  [T1,n1,job=SCHEMA CHANGE GC id=848634212708319233] 2962  updated progress payload: tables:<id:113 status:WAITING_FOR_MVCC_GC > tables:<id:116 status:WAITING_FOR_MVCC_GC > ranges_unsplit_done:true
I230317 11:36:49.073337 98618 sql/gcjob/table_garbage_collection.go:208  [T1,n1,job=SCHEMA CHANGE GC id=848634212798824449] 2963  deleting data for table 114
I230317 11:36:49.075496 98618 sql/gcjob/table_garbage_collection.go:208  [T1,n1,job=SCHEMA CHANGE GC id=848634212798824449] 2964  deleting data for table 118
I230317 11:36:49.078675 98416 jobs/registry.go:1492  [T1,n1] 2965  NEW SCHEMA CHANGE job 848634212651204609: stepping through state succeeded
I230317 11:36:49.081859 98463 sql/gcjob/gc_job_utils.go:296  [T1,n1,job=SCHEMA CHANGE GC id=848634212708319233] 2966  updated running status: waiting for MVCC GC
I230317 11:36:49.083915 98618 sql/gcjob/gc_job_utils.go:289  [T1,n1,job=SCHEMA CHANGE GC id=848634212798824449] 2967  updated progress payload: tables:<id:114 status:WAITING_FOR_MVCC_GC > tables:<id:118 status:WAITING_FOR_MVCC_GC > ranges_unsplit_done:true
I230317 11:36:49.084061 98511 jobs/adopt.go:247  [T1,n1] 2968  job 102: resuming execution
I230317 11:36:49.105918 98463 sql/gcjob/gc_job.go:378  [T1,n1,job=SCHEMA CHANGE GC id=848634212708319233] 2970  not waiting for MVCC GC in /Table/113 due to testing knob
I230317 11:36:49.105993 98463 sql/schema_changer.go:3036  [T1,n1,job=SCHEMA CHANGE GC id=848634212708319233] 2971  removing table descriptor and zone config for table 113
I230317 11:36:49.088998 98783 jobs/registry.go:1492  [T1,n1] 2969  SCHEMA CHANGE GC job 848634213013553153: stepping through state running
I230317 11:36:49.108694 98618 sql/gcjob/gc_job_utils.go:296  [T1,n1,job=SCHEMA CHANGE GC id=848634212798824449] 2972  updated running status: waiting for MVCC GC
I230317 11:36:49.110016 98463 sql/gcjob/gc_job.go:378  [T1,n1,job=SCHEMA CHANGE GC id=848634212708319233] 2973  not waiting for MVCC GC in /Table/116 due to testing knob
I230317 11:36:49.110085 98463 sql/schema_changer.go:3036  [T1,n1,job=SCHEMA CHANGE GC id=848634212708319233] 2974  removing table descriptor and zone config for table 116
E230317 11:36:49.110618 98511 jobs/adopt.go:211  [T1,n1] 2975  could not run claimed job 102: no resumer is available for AUTO CONFIG RUNNER
I230317 11:36:49.111018 98618 sql/gcjob/gc_job.go:378  [T1,n1,job=SCHEMA CHANGE GC id=848634212798824449] 2976  not waiting for MVCC GC in /Table/114 due to testing knob
I230317 11:36:49.111089 98618 sql/schema_changer.go:3036  [T1,n1,job=SCHEMA CHANGE GC id=848634212798824449] 2977  removing table descriptor and zone config for table 114
I230317 11:36:49.113637 98463 jobs/registry.go:1492  [T1,n1] 2978  SCHEMA CHANGE GC job 848634212708319233: stepping through state succeeded
I230317 11:36:49.121526 92183 jobs/wait.go:142  [T1,n1,client=127.0.0.1:51116,user=root] 2979  waited for 1 [848634212651204609] queued jobs to complete 225.845534ms
W230317 11:36:49.122334 90932 jobs/registry.go:1052  [T1,n1] 2980  canceling all adopted jobs due to stopper quiescing
I230317 11:36:49.122480 90551 server/start_listen.go:103  [T1,n1] 2981  server shutting down: instructing cmux to stop accepting
W230317 11:36:49.122645 90898 sql/sqlliveness/slinstance/slinstance.go:312  [n1] 2982  exiting heartbeat loop
W230317 11:36:49.122636 98947 kv/kvserver/intentresolver/intent_resolver.go:826  [-] 2983  failed to gc transaction record: could not GC completed transaction anchored at /Table/3/1/116/2/1: node unavailable; try another peer
W230317 11:36:49.122666 90898 sql/sqlliveness/slinstance/slinstance.go:297  [n1] 2984  exiting heartbeat loop
E230317 11:36:49.122691 90898 server/server_sql.go:490  [n1] 2985  failed to run update of instance with new session ID: node unavailable; try another peer
I230317 11:36:49.122889 90894 sql/stats/automatic_stats.go:507  [T1,n1] 2986  quiescing auto stats refresher
I230317 11:36:49.123394 91844 jobs/registry.go:1492  [T1,n1] 2987  KEY VISUALIZER job 100: stepping through state succeeded
E230317 11:36:49.123491 91844 jobs/registry.go:924  [T1,n1] 2988  error getting live session: node unavailable; try another peer
I230317 11:36:49.124029 92732 rpc/context.go:2249  [T1,n1,rnode=1,raddr=127.0.0.1:40075,class=default,rpc] 2989  connection heartbeat loop ended with err: <nil>
    schema_changes_in_parallel_test.go:76: 
        	Error Trace:	github.com/cockroachdb/cockroach/pkg/sql/tests/schema_changes_in_parallel_test.go:76
        	Error:      	Received unexpected error:
        	            	pq: failed to ingest index entries during backfill: not enough memory available to create a BulkAdder: root: memory budget exceeded: 33554432 bytes requested, 102803456 currently allocated, 134217728 bytes in budget
        	Test:       	TestSchemaChangesInParallel
I230317 11:36:49.124926 91046 jobs/registry.go:1492  [T1,n1] 2990  AUTO SPAN CONFIG RECONCILIATION job 848634201743982593: stepping through state succeeded
E230317 11:36:49.124976 90934 jobs/registry.go:1036  [T1,n1] 2991  error processing claimed jobs: could not query for claimed jobs: select-running/get-claimed-jobs: failed to read query result: query execution canceled
--- FAIL: TestSchemaChangesInParallel (3.84s)
```
<p>Parameters: <code>TAGS=bazel,gss</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSchemaChangesInParallel.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-25550

--- a/pkg/sql/tests/schema_changes_in_parallel_test.go
+++ b/pkg/sql/tests/schema_changes_in_parallel_test.go
@@ -39,6 +39,7 @@ func TestSchemaChangesInParallel(t *testing.T) {
 			},
 			JobsTestingKnobs: jobs.NewTestingKnobsWithShortIntervals(),
 		},
+		SQLMemoryPoolSize: 1 << 30, /* 1GiB */
 	})
 	defer s.Stopper().Stop(ctx)
 

